basket-analysis/basket-analysis-quickstart/Basket_Analysis.ipynb

```diff
@@ -0,0 +1,709 @@
+   "metadata": {
+    "collapsed": false,
+    "name": "cell31"
+   },
+   "source": [
+    "Once the session is started, we can project our node and relationship views into a GDS in-memory graph. The graph will be identified by the name \"parts_in_orders\".\n",
+    "\n",
+    "* The mandatory parameters are the node tables and the relationship tables.\n",
+    "* A node table mapping points from a table/view to a node label that is used in the GDS graph.\n",
+    "* The name of node label is based on the table/view name used in the projection, and case is preserved.\n",
+    "For example, the rows of 'tpch_example.gds.Part' will be nodes labeled as 'Part'.\n",
+    "* Relationship tables need a bit more configuration.\n",
+    "We need to specify source and target tables.\n",
+    "* The relationships are represented as typed relationships is the GDS graph, where similarly to nodes, the table/view name is taken as the relationship type.\n",
+    "* For example, 'tpch_example.gds.part_in_order' below gives rise to the relationship 'part_in_order' in the GDS graph.\n",
+    "* We also specify the optional read concurrency to optimize building the graph projection.\n",
+    "* The concurrency can be set to the number of cores available on the compute pool node."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "786748d8-9d02-4076-8a07-65afdd67e757",
+   "metadata": {
+    "collapsed": false,
+    "name": "cell32"
+   },
+   "source": [
+    "The graph we project is a so-called bipartite graph, as it contains two types of nodes and all relationships point from one type to the other.\n",
+    "The node similarity algorithm looks at all pairs of nodes of the first type and calculates the similarity for each pair based on common relationships.\n",
+    "In our case, the algorithm will calculate the similarity between two parts based on the orders in which they appear.\n",
+    "The algorithm produces new relationships between parts, the relationship property is the similarity score.\n",
+    "For further information on the node similarity algorithm, please refer to the [GDS documentation](https://neo4j.com/docs/graph-data-science/current/algorithms/node-similarity/)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "a441d0d6-0a22-4fcb-b82d-00a0f83d3990",
+   "metadata": {
+    "collapsed": false,
+    "name": "cell33"
+   },
+   "source": [
+    "Once the algorithm has finished, we can write the results back to Snowflake tables for further analysis.\n",
+    "We want to write back the similarity relationships between parts. \n",
+    "The specified table will contain the original source and target node ids and the similarity score."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "2247dfa5-abcf-482b-91b2-9ce38840a3f6",
+   "metadata": {
+    "language": "sql",
+    "name": "cell34"
+   },
+   "outputs": [],
+   "source": [
+    "CALL neo4j_graph_analytics.graph.node_similarity('CPU_X64_L', {\n",
```

Comment:
I think this notebook mirrors this file https://github.com/neo4j-product-examples/snowflake-graph-analytics/blob/main/product-recommendation/product-recommendation.sql

please have a look, you can find something helpful there